flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxyV2.java

```diff
@@ -28,16 +51,210 @@
+		this.fanOutRecordPublisherConfiguration = new FanOutRecordPublisherConfiguration(configProps, streams);
+	}
+
+	/**
+	 * Creates a Kinesis proxy V2.
+	 *
+	 * @param configProps configuration properties
+	 * @param streams list of kinesis stream names
+	 * @return the created kinesis proxy v2
+	 */
+	public static KinesisProxyV2Interface create(Properties configProps, List<String> streams) {
+		return new KinesisProxyV2(configProps, streams);
+	}
+
+	/**
+	 * Create the Kinesis client, using the provided configuration properties.
+	 * Derived classes can override this method to customize the client configuration.
+	 *
+	 * @param configProps the properties map used to create the Kinesis Client
+	 * @return a Kinesis Client
+	 */
+	protected KinesisAsyncClient createKinesisAsyncClient(final Properties configProps) {
+		final ClientConfiguration config = new ClientConfigurationFactory().getConfig();
+		return AwsV2Util.createKinesisAsyncClient(configProps, config);
+	}
+
+	/**
+	 * {@inheritDoc}
 	 */
-	public KinesisProxyV2(final KinesisAsyncClient kinesisAsyncClient) {
-		this.kinesisAsyncClient = kinesisAsyncClient;
+	@Override
+	public Map<String, String> describeStream(List<String> streams) throws InterruptedException, ExecutionException {
+		Map<String, String> result = new HashMap<>();
+		for (String stream : streams) {
+			DescribeStreamRequest describeStreamRequest = DescribeStreamRequest
+				.builder()
+				.streamName(stream)
+				.build();
+			DescribeStreamResponse describeStreamResponse = null;
+
+			int retryCount = 0;
+			while (retryCount <= fanOutRecordPublisherConfiguration.getDescribeStreamMaxRetries() && describeStreamResponse == null) {
+				try {
+					describeStreamResponse = kinesisAsyncClient.describeStream(describeStreamRequest).get();
+				} catch (ExecutionException ex) {
+					if (AwsV2Util.isRecoverableException(ex)) {
+						long backoffMillis = fullJitterBackoff(
+							fanOutRecordPublisherConfiguration.getDescribeStreamBaseBackoffMillis(), fanOutRecordPublisherConfiguration.getDescribeStreamMaxBackoffMillis(), fanOutRecordPublisherConfiguration.getDescribeStreamExpConstant(), retryCount++);
+						LOG.warn("Got recoverable AmazonServiceException when trying to describe stream " + stream + ". Backing off for "
+							+ backoffMillis + " millis (" + ex.getClass().getName() + ": " + ex.getMessage() + ")");
+						Thread.sleep(backoffMillis);
+					} else {
+						throw ex;
+					}
+				}
+			}
+			if (describeStreamResponse == null) {
+				throw new RuntimeException("Retries exceeded for describeStream operation - all " + fanOutRecordPublisherConfiguration.getDescribeStreamMaxRetries() +
+					" retry attempts failed.");
+			}
+			result.put(stream, describeStreamResponse.streamDescription().streamARN());
+		}
+		return result;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<FanOutStreamInfo> registerStreamConsumer(Map<String, String> streamArns) throws InterruptedException, ExecutionException {
```

Comment:
@xiaolong-sn again I think this should accept a single streamArn/consumerName and loop externally

Reply:
I changed the Interface accordingly.